test/bulk-update/reporter/excel-reporter.test.js

```diff
@@ -0,0 +1,156 @@
+import { expect } from '@esm-bundle/chai';
+import fs from 'fs';
+import xlsx from 'xlsx';
+import sinon from 'sinon';
+import ExcelReporter from '../../../bulk-update/reporter/excel-reporter.js';
+
+const { pathname } = new URL('.', import.meta.url);
+
+const deleteObjectProperty = (obj, prop) => Object.keys(obj).forEach((key) => {
+  delete obj[key][prop];
+});
+
+describe('ExcelReporter', () => {
+  describe('Check sheets js library is called', () => {
+    const sandbox = sinon.createSandbox();
+
+    beforeEach(() => {
+      sandbox.spy(xlsx.utils);
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it('should create a new workbook', () => {
+      const reporter = new ExcelReporter();
+
+      expect(reporter).is.not.undefined;
+      expect(xlsx.utils.book_new.calledOnce).to.be.true;
+    });
+
+    it('should append log to sheet', () => {
+      const reporter = new ExcelReporter();
+
+      reporter.log('topic', 'status', 'message', 'arg1', 'arg2');
+
+      expect(xlsx.utils.book_append_sheet.calledTwice).to.be.true;
+    });
+  });
+
+  describe('Check report', () => {
+    it('should log a message', () => {
+      const reporter = new ExcelReporter();
+
+      reporter.log('topic', 'status', 'message', 'arg1', 'arg2');
+
+      const report = reporter.getReport();
+
+      expect(report.logs.topic).to.have.lengthOf(1);
+      expect(report.logs.topic[0]).to.deep.equal({
+        status: 'status',
+        message: 'message',
+        0: 'arg1',
+        1: 'arg2',
+      });
+    });
+
+    it('should calculate the totals for each topic and status', () => {
```

Comment:
This test is a duplicate of line 25 in `consoleReporter`. Though it makes sure the `calculateTotals` method in the ExcelReporter logs correctly, since they both extend `BaseReporter` you could probably rely on the output of this same test that is in `BaseReporter` for this verification, and use these test to check for differences in the method. 

For example, something like: 
`expect(xlsx.utils.sheet_add_aoa.calledOnce).to.be.true`

Reply:
Created a new test to check `aoa_to_sheet` and `sheet_add_aoa`.